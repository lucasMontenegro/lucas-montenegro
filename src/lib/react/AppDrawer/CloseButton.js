import React from "react"
import Tooltip from "@material-ui/core/Tooltip"
import IconButton from "@material-ui/core/IconButton"
import { FontAwesomeIcon } from "lib/react/fontAwesome"
export default function CloseButton (props) {
  return props.isDesktop ? null : (
    <Tooltip
      title={props.t({
        en: () => `Close drawer`,
        es: () => `Cerrar panel lateral`,
      })}
    >
      <IconButton
        edge="start"
        className="icon-button"
        aria-controls="lib-react-app_drawer"
        onClick={props.onClick}
      >
        <FontAwesomeIcon icon={[`fas`, `arrow-left`]} />
      </IconButton>
    </Tooltip>
  )
}