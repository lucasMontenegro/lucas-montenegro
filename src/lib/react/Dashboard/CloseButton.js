import React from "react"
import Tooltip from "@material-ui/core/Tooltip"
import IconButton from "@material-ui/core/IconButton"
import { FontAwesomeIcon } from "lib/react/fontAwesome"
export default function CloseButton (props) {
  return props.isDesktop ? null : (
    <Tooltip
      title={props.t({
        en: () => `Close dashboard`,
        es: () => `Cerrar panel de control`,
      })}
    >
      <IconButton
        edge="start"
        className="icon-button"
        aria-controls="lib-react-dashboard"
        onClick={props.onClick}
      >
        <FontAwesomeIcon icon={[`fas`, `arrow-left`]} />
      </IconButton>
    </Tooltip>
  )
}