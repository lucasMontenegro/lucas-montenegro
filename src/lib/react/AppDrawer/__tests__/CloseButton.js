import IconButton from "@material-ui/core/IconButton"
import renderer from "react-test-renderer"
import React from "react"
import CloseButton from "../CloseButton"
jest.mock(`@material-ui/core/Tooltip`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <div {...props} className="Tooltip" /> }
})
jest.mock(`@material-ui/core/IconButton`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    default: jest.fn(props => (
      <button {...props} className="IconButton" onClick={props.onClick()} />
    )),
  }
})
jest.mock(`@material-ui/icons/Close`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <span {...props}>CloseIcon</span> }
})
describe(`../CloseButton`, () => {
  it(`should use the right versions of its dependencies`, () => {
    expect(jestUtils.getDependencies([
      `react`,
      `@material-ui/core`,
      `@material-ui/icons`,
      `prop-types`,
    ])).toMatchSnapshot()
  })
  describe(`<CloseButton /> (is desktop true)`, () => {
    it(`should render`, () => {
      expect(renderer.create(
        <CloseButton isDesktop={true} t={() => {}} onClick={() => {}} />
      ).toJSON()).toBeNull()
    })
  })
  describe(`<CloseButton /> (is desktop false)`, () => {
    it(`should render`, () => {
      function t (source) {
        return Object.keys(source).reduce((translations, languageCode) => {
          translations[languageCode] = source[languageCode]()
          return translations
        }, {})
      }
      expect(renderer.create(
        <CloseButton isDesktop={false} t={t} onClick={() => `props.onClick()`} />
      ).toJSON()).toMatchSnapshot()
    })
  })
})