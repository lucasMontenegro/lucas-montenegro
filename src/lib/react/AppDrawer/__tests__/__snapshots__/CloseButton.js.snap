// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../CloseButton <CloseButton /> (is desktop false) should render 1`] = `
<div
  className="Tooltip"
  title={
    Object {
      "en": "Close drawer",
      "es": "Cerrar panel lateral",
    }
  }
>
  <button
    aria-controls="lib-react-app_drawer"
    className="IconButton"
    edge="start"
    onClick="props.onClick"
  >
    <i
      className="FontAwesomeIcon"
      icon={
        Array [
          "fas",
          "arrow-left",
        ]
      }
    />
  </button>
</div>
`;

exports[`../CloseButton should use the right versions of its dependencies 1`] = `
Object {
  "@material-ui/core": "^4.5.0",
  "@material-ui/icons": "^4.4.3",
  "prop-types": "^15.7.2",
  "react": "^16.10.1",
}
`;
